browser/components/newtab/SearchProvider.jsm

```diff
@@ -167,24 +167,27 @@ this.SearchProvider = {
       controller.remoteTimeout = data.remoteTimeout || undefined;
       let isPrivate = PrivateBrowsingUtils.isBrowserPrivate(browser);
 
-      // fetch() rejects its promise if there's a pending request, but since we
-      // process our event queue serially, there's never a pending request.
-      let suggestions = yield controller.fetch(data.searchString, isPrivate, engine);
-
-      if (!suggestions) {
-        throw new Error("There are no suggestions");
-      }
+      let suggestion;
+      try {
+        // fetch() rejects its promise if there's a pending request, and due to it's
+        // asynchronous nature, it is possible that fetch() rejects and throws with a
+        // "HTTP request aborted". In that case, the suggestion is undefined.
+        let suggestions = yield controller.fetch(data.searchString, isPrivate, engine);
```

Comment:
one is suggestion (singular, the one we want to return) and one is suggestions (the set of all suggestions available)